src/main/java/com/dataguise/saas/service/SchedulerService.java

```diff
@@ -0,0 +1,434 @@
+        }
+        try {
+            taskSchedulerTaskHistBeanList = restDriver.getTaskInstancesBySchedule(scheduleId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return taskSchedulerTaskHistBeanList;
+    }
+
+    /**
+     * Get schedule summary by scheduleId
+     *
+     * @param scheduleId
+     */
+    public DgTaskScheduledExecutionsBean getScheduleSummary(Optional<Integer> scheduleId)
+            throws BadRequestException, BadGatewayException {
+        DgTaskScheduledExecutionsBean scheduleSummary = new DgTaskScheduledExecutionsBean();
+        try {
+            if (scheduleId.isPresent() && scheduleId.get() > 0) {
+                scheduleSummary = restDriver.getScheduleSummary(scheduleId.get());
+            } else {
+                scheduleSummary = restDriver.getScheduleSummary(0);
+            }
+        } catch (DgDrvException e) {
+            logger.error("Error retrieving the schedule summary"); // Not throwing an error in this case because GUI require default values
+            Long defaultSummary = new Long(0);
+            scheduleSummary.setFailedExecutions(defaultSummary);
+            scheduleSummary.setInProgressExecutions(defaultSummary);
+            scheduleSummary.setSuccessfulExecutions(defaultSummary);
+            scheduleSummary.setTotalDetectionScheduler(defaultSummary);
+            scheduleSummary.setTotalExecutions(defaultSummary);
+            scheduleSummary.setTotalMaskingScheduler(defaultSummary);
+        }
+        return scheduleSummary;
+    }
+
+
+    public SchedulerDTO formatScheduleDTO(SchedulerDTO schedulerDto) throws BadRequestException, InternalServerException, BadGatewayException {
+        String startDateTime = null;
+        String weekday = null;
+        List<String> weekdays = new ArrayList<>();
+        java.util.Date localZoneDate = null;
+
+        //TODO: Accept date in ISO 8601 format and convert it into controller readable format.
+        String isoInputDate = schedulerDto.getStartDateTime();
+        logger.info("Input ISO date from UI: " + isoInputDate);
+
+        try {
+            localZoneDate = Date.from(ZonedDateTime.parse(isoInputDate).toInstant());
+            logger.info("Local time zone of UI:" + localZoneDate.toString()); //ISO TO LOCAL TIME ZONE
+        } catch (DateTimeParseException e) {
+            throw new InternalServerException(ErrorConstants.ERROR_PARSE_ISO_DATE.getErrorMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        SimpleDateFormat dof = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
+        SimpleDateFormat dof1 = new SimpleDateFormat("EEEE");
+
+        TimeZone tz = TimeZone.getTimeZone("UTC");
+        dof.setTimeZone(tz);
+        dof1.setTimeZone(tz);
+
+        startDateTime = dof.format(localZoneDate);
+        logger.info("FINAL CORRECT DATE: " + startDateTime);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/SchedulerService.java L397
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1540283877>*